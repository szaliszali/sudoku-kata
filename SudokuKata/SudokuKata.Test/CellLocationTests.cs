namespace SudokuKata.Test;

internal class CellLocationTests
{
    [Test]
    public void BlockIndex()
    {
        var board = new Board();
        Approvals.VerifyAll(Enumerable.Range(0, 81).Select(i => new CellLocation(board, i / 9, i % 9).BlockIndex()), "");
    }

    [Test]
    public void BlockRow()
    {
        var board = new Board();
        Approvals.VerifyAll(Enumerable.Range(0, 81).Select(i => new CellLocation(board, i / 9, i % 9).BlockRow()), "");
    }

    [Test]
    public void BlockCol()
    {
        var board = new Board();
        Approvals.VerifyAll(Enumerable.Range(0, 81).Select(i => new CellLocation(board, i / 9, i % 9).BlockCol()), "");
    }
}
