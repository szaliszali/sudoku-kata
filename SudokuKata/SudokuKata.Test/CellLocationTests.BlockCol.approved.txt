[0] = 0
[1] = 0
[2] = 0
[3] = 1
[4] = 1
[5] = 1
[6] = 2
[7] = 2
[8] = 2
[9] = 0
[10] = 0
[11] = 0
[12] = 1
[13] = 1
[14] = 1
[15] = 2
[16] = 2
[17] = 2
[18] = 0
[19] = 0
[20] = 0
[21] = 1
[22] = 1
[23] = 1
[24] = 2
[25] = 2
[26] = 2
[27] = 0
[28] = 0
[29] = 0
[30] = 1
[31] = 1
[32] = 1
[33] = 2
[34] = 2
[35] = 2
[36] = 0
[37] = 0
[38] = 0
[39] = 1
[40] = 1
[41] = 1
[42] = 2
[43] = 2
[44] = 2
[45] = 0
[46] = 0
[47] = 0
[48] = 1
[49] = 1
[50] = 1
[51] = 2
[52] = 2
[53] = 2
[54] = 0
[55] = 0
[56] = 0
[57] = 1
[58] = 1
[59] = 1
[60] = 2
[61] = 2
[62] = 2
[63] = 0
[64] = 0
[65] = 0
[66] = 1
[67] = 1
[68] = 1
[69] = 2
[70] = 2
[71] = 2
[72] = 0
[73] = 0
[74] = 0
[75] = 1
[76] = 1
[77] = 1
[78] = 2
[79] = 2
[80] = 2
